### DISCUSSION TITLE
Multiple shadows

### DISCUSSION THREAD

**Discussion by EvanKoe** [yshui/picom#965] [Q&A]
Hello !
I started a new neomorphism rice using i3-gaps and picom, and I was wondering if we could create more than one shadow on windows using picom. I need two shadows : one black on bottom right et one white on top left.  
[Here is an example](https://dribbble.com/shots/9890709/attachments/1924410?mode=media)  
I didn't find anything about multiple shadows on the documentation, that's why I ask here.   
  
Do you know if this is possible ? If no, don't you think it could be a great feature ?   
Thanks !

**Comment by nick87720z** [comment]
> Do you know if this is possible ?

Why not? Patches are welcome :)
Closer to implementation details - if shadows are done with shader, than there could be another place for custom shader option.
In the end - I started to think about option for entire custom directory with predefined shader files (e.g. "window-shader", "screen-shader", "shadow-shader"...).

> If no, don't you think it could be a great feature ?

Can't find practical value, though perhaps it's bound to how you plan total desktop look. As for pure eye-candy - I would use shadow shader for things like new year lights or so :) *(it could be even animated)*
